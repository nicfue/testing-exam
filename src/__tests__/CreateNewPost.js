import React from 'react';
import { mount } from 'enzyme';
import CreateNewPost from '../components/CreateNewPost';


describe('testing create new post', () => {
  const newPost = jest.fn();
  const wrapper = mount(<CreateNewPost author="" updatePosts={newPost} />);

  it('should create a new title', ()=>{
    const title = wrapper.find("#title");
    title.simulate('change', {target: { name: 'title', value: 'testTitle'}});
    expect(wrapper.state().title).toEqual('testTitle');
    });

  it('should create new content', ()=>{
    const content = wrapper.find("#content")
    content.simulate('change', {target: { name: 'content', value: 'testContent'}});
    expect(wrapper.state().content).toEqual('testContent');
    });

  it('testing submit', () => {
    const submit = wrapper.find("form")
    submit.simulate('submit')
  })

  it('snapshots when match', ()=> {
    expect(wrapper).toMatchSnapshot();
  })

})
