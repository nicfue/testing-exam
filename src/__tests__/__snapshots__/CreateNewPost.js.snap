// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing create new post snapshots when match 1`] = `
<CreateNewPost
  author=""
  updatePosts={[Function]}
>
  <form
    className="container mx-auto flex flex-col p-6"
    onSubmit={[Function]}
  >
    <label
      className="mb-2"
      htmlFor="title"
    >
      Title
    </label>
    <input
      className="shadow rounded border mb-2 p-2"
      id="title"
      name="title"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      className="mb-2"
      htmlFor="content"
    >
      Content 
    </label>
    <textarea
      className="shadow rounded border mb-2 p-2"
      id="content"
      name="content"
      onChange={[Function]}
      value=""
    />
    <input
      className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
      type="submit"
      value="Create"
    />
  </form>
</CreateNewPost>
`;
